platforms/nuttx/src/bootloader/bl.c

```diff
@@ -142,8 +142,8 @@
 #define STATE_ALLOWS_REBOOT       (STATE_PROTO_GET_SYNC)
 #  define SET_BL_STATE(s)
 #else
-#define STATE_ALLOWS_ERASE        (STATE_PROTO_GET_SYNC|STATE_PROTO_GET_DEVICE)
-#define STATE_ALLOWS_REBOOT       (STATE_ALLOWS_ERASE|STATE_PROTO_PROG_MULTI|STATE_PROTO_GET_CRC)
+#define STATE_ALLOWS_ERASE        (STATE_PROTO_GET_SYNC)
```

Comment:
This is the code as it is used on all the non H7 board.  The STATE_PROTO_GET_CHIP was removed. But QGC should not erase a device before it knows what it is. Using STATE_PROTO_GET_DEVICE

https://github.com/PX4/Bootloader/commit/d83169d50c333ab33d8206a8a29f6efc9605c39b#diff-819bdaaa83fd1bf4f350cfb063101f05


This is not a good idea.  The protection is defeated and the board can be erased by accident.